### PR TITLE
 golang filter: implement HeaderMap.Add

### PR DESCRIPTION
**Pull request by RRobot-lm**
Commit Message: golang filter: implement HeaderMap.Add
Additional Description:
Risk Level: Low
Testing: contrib/golang/filters/http/test/golang_integration_test.cc
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: contrib/golang/filters/http/source/golang_filter.cc [hunk 1]
```diff
@@ -639,7 +639,20 @@ CAPIStatus Filter::setHeader(absl::string_view key, absl::string_view value) {
...
+  case HeaderAdd:
+    headers_->addCopy(Http::LowerCaseString(key), value);
+    break;
+
+  case HeaderSet:
+    headers_->setCopy(Http::LowerCaseString(key), value);
+    break;
+
+  default:
+    ENVOY_LOG(error, "unknown header action {}, ignored", act);
```
Comment (doujiang24) [inline contrib/golang/filters/http/source/golang_filter.cc:653]: better `RELEASE_ASSERT` here, we assume the Go code should be trusted.

**Review by doujiang24 — COMMENTED** [review]
just a tiny suggestion, otherwise, LGTM.

**Review by doujiang24 — DISMISSED** [review]

**Review by mattklein123 — APPROVED** [review]

**Comment by mattklein123** [comment]
Oops looks like needs a main merge.

/wait

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**